File: app/src/main/java/java/android/notes/wrapper/LaunchFragment.java
package java.android.notes.wrapper;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import java.android.notes.R;
import java.android.notes.wrapper.alerts.CustomDialogFragmentWithView;
import java.android.notes.wrapper.alerts.CustomDialogListener;
import java.android.notes.wrapper.helpers.CreateFragment;

public class LaunchFragment extends Fragment implements View.OnClickListener, CustomDialogListener {
    private CustomDialogFragmentWithView dialog;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        setHasOptionsMenu(true);

        return inflater.inflate(R.layout.fragment_launch,container,false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        ((AppCompatActivity) requireActivity()).getSupportActionBar().hide();
        //visible();  // временно

        view.findViewById(R.id.buttonOpenNotes).setOnClickListener(this);
        view.findViewById(R.id.buttonClose).setOnClickListener(this);
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
        super.onCreateOptionsMenu(menu, inflater);

        MenuItem item = menu.findItem(R.id.action_exit);
        if(item!=null){
            item.setVisible(false);
        }

    }

    @Override
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.buttonOpenNotes:

                //notVisible(); // временно

                dialog = new CustomDialogFragmentWithView();
                dialog.setListener(this);
                dialog.show(requireActivity().getSupportFragmentManager(),CustomDialogFragmentWithView.TAG);

                break;
            case R.id.buttonClose:
                requireActivity().finish();
                break;
        }
    }

    ////////////////////////
    public void visible(){
        requireActivity().getSupportFragmentManager().beginTransaction().show(this).commit();
    }

    public void notVisible(){
        requireActivity().getSupportFragmentManager().beginTransaction().hide(this).commit();
    }

    @Override
    public void onOk() {
        if(dialog.editText.getText().toString().equals("4")){
            CreateFragment.createNotesFragment( (AppCompatActivity)requireActivity() );
        }else{
            Toast.makeText((AppCompatActivity)requireActivity(),"Результат проведённого тэста указывает на то, что уровня знаний недостаточно. Совершенствуйте свои навыки и обязательно возвращайтесь снова", Toast.LENGTH_LONG).show();
        }

    }

    @Override
    public void onNo() {
        Toast.makeText((AppCompatActivity)requireActivity(), "Действие отменено пользователем", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/java/android/notes/wrapper/fragments/extra/LaunchFragment.kt
package java.android.notes.wrapper.fragments.extra

import java.android.notes.wrapper.alerts.CustomDialogListener
import java.android.notes.wrapper.alerts.CustomDialogFragmentWithView
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import java.android.notes.R
import java.android.notes.wrapper.helpers.CreateFragment
import androidx.appcompat.app.AppCompatActivity
import android.widget.Toast
import androidx.fragment.app.Fragment

class LaunchFragment : Fragment(), View.OnClickListener, CustomDialogListener {
    private val dialog: CustomDialogFragmentWithView? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        setHasOptionsMenu(true)
        return inflater.inflate(R.layout.fragment_launch, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        view.findViewById<View>(R.id.buttonOpenNotes).setOnClickListener(this)
        view.findViewById<View>(R.id.buttonClose).setOnClickListener(this)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.buttonOpenNotes ->
                //dialog = new CustomDialogFragmentWithView();
                //dialog.setListener(this);
                //dialog.show(requireActivity().getSupportFragmentManager(),CustomDialogFragmentWithView.TAG);

                // временно
                CreateFragment.createNotesFragment(requireActivity() as AppCompatActivity)
            R.id.buttonClose -> requireActivity().finish()
        }
    }

    ////////////////////////
    override fun onOk() {
        if (dialog!!.editText!!.text.toString() == "4") {
            CreateFragment.createNotesFragment(requireActivity() as AppCompatActivity)
        } else {
            Toast.makeText(
                requireActivity() as AppCompatActivity,
                "Результат проведённого тэста указывает на то, что уровня знаний недостаточно. Совершенствуйте свои навыки и обязательно возвращайтесь снова",
                Toast.LENGTH_LONG
            ).show()
        }
    }

    override fun onNo() {
        Toast.makeText(
            requireActivity() as AppCompatActivity,
            "Действие отменено пользователем",
            Toast.LENGTH_SHORT
        ).show()
    }
}